### PR TITLE
ENH: add support for NEP 35 array constructors (`require`, `identiy`, `eye`, `tri`, `genfromtxt` and `loadtxt`) with `like=Quantity(...))`

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Concludes the first half of #17124 (`units`)

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/whatsnew/7.0.rst [hunk 1]
```diff
@@ -86,6 +87,61 @@ Example:
...
+.. code-block:: python
+
+    >>> import numpy as np
+    >>> from astropy import units as u
+    >>> q = u.Quantity(1.0, u.m)
+    >>> np.arange(0, 10, 1, like=q)
+    <Quantity [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] m>
+    >>> np.eye(3, like=q)
+    <Quantity [[1., 0., 0.],
+            [0., 1., 0.],
```
Comment (mhvk) [inline docs/whatsnew/7.0.rst:109]: Alignment? Here and below

**Review by mhvk — COMMENTED** [review]
Thanks, great to see this wrapped up! I'm not 100% sure we can fully support `require` (in particular "OWNDATA"), but to be honest, I'd never even heard about the function, so think it is sufficiently obscure not to worry... (Also, probably just a consequence of #7244.)

File: docs/whatsnew/7.0.rst (hunk 1 above)
Comment (neutrinoceros) [reply docs/whatsnew/7.0.rst]: thanks. Fixed !

**Review by mhvk — APPROVED** [review]
OK, thanks for the quick fix! Will set it to merge...

**Pull request merged**